[bbbbcccd] iP

bbbbcccd opened the pull request:
# LevelHundred
>“Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

LevelHundred frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2425s1.github.io/website/admin/ip-w4.html).
double-click it.
2. add your tasks.
3. let it manage your tasks for you 😉
And it is **FREE**!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

bbbbcccd committed:
Rename chat bot

bbbbcccd committed:
Implement an initial skeletal version of chatbot

- Greet user
- Exit

bbbbcccd committed:
Finish Level 1. Echo

- Echo commands entered by user
- Exits when user types command bye

bbbbcccd committed:
Shift greet and exit to Ui

bbbbcccd committed:
Create Storage and Task classes

bbbbcccd committed:
Complete Level 2

- Store task entered by user
- List all tasks

bbbbcccd committed:
Add feature to mark task as done / undone

bbbbcccd committed:
Complete Level 3

- Add feature to mark tasks as done / undone
- Add exception handling for invalid commands

bbbbcccd committed:
Create subclasses of Task

bbbbcccd committed:
Add feature to create Todo, Deadline and Task

bbbbcccd committed:
Update UI when a new task is added

bbbbcccd committed:
Add automated text UI testing

bbbbcccd committed:
Add exceptions

bbbbcccd committed:
Throw exceptions when creating tasks

bbbbcccd committed:
Remove todo

bbbbcccd committed:
Add exceptions for todo command

bbbbcccd committed:
Fix seperation of concerns issue

bbbbcccd committed:
Add support for deleting tasks from list

bbbbcccd committed:
Add exception for mark, unmark and delete

- InvalidArgumentException if user inputs multi-word task index

bbbbcccd committed:
Add exception for invalid commands

bbbbcccd committed:
Create file to store tasks

bbbbcccd committed:
Separate Storage and TaskList class

bbbbcccd committed:
Create Parser class

bbbbcccd committed:
Update File format

bbbbcccd committed:
Create new Exception for invalid storage file

bbbbcccd committed:
Update load function

bbbbcccd committed:
Remove unused imports

bbbbcccd committed:
Fix bug: close scanner

bbbbcccd committed:
Create new file if not existing

bbbbcccd committed:
Throw exceptions if file format is invalid

bbbbcccd committed:
Load tasks from storage into taskList

bbbbcccd committed:
Update storage file whenever task list changes

bbbbcccd committed:
Fix bug: Add break to delete case

bbbbcccd committed:
Handle file / directory not existing

bbbbcccd committed:
Merge branch 'branch-level-7' into master

bbbbcccd committed:
Create function to parse user date

bbbbcccd committed:
Parse dates from user input when adding task

bbbbcccd committed:
Ignore storage file

bbbbcccd committed:
Add command classes to execute user command

bbbbcccd committed:
Add parsing method for user commands

bbbbcccd committed:
Parse user commands using Parser and UserCommands

bbbbcccd committed:
Remove unused imports

bbbbcccd committed:
Create comand package

bbbbcccd committed:
Add exception package

bbbbcccd committed:
Add task package

bbbbcccd committed:
Add ui package

bbbbcccd committed:
Add utility pacakge

bbbbcccd committed:
Remove old files

bbbbcccd committed:
Add levelhundred package

bbbbcccd committed:
Merge branch 'master' into branch-a-packages

bbbbcccd committed:
Merge branch 'add-gradle-support'

bbbbcccd committed:
Update build file with entry point + checkstyle

bbbbcccd committed:
Add checkstyle

bbbbcccd committed:
Add test for UserCommand

bbbbcccd committed:
Reorder imports to follow checkstyle

bbbbcccd committed:
Add test for Parser

bbbbcccd committed:
Rename shadowJar file

bbbbcccd committed:
Add JavaDocs for command package

bbbbcccd committed:
Add JavaDocs for exception package

bbbbcccd committed:
Add JavaDocs for task package

bbbbcccd committed:
Add JavaDocs for ui package

bbbbcccd committed:
Add JavaDocs for utility package

bbbbcccd committed:
Add JavaDocs for levelhundred package

bbbbcccd committed:
Suppress whitespace and eof line

bbbbcccd committed:
Fix style

bbbbcccd committed:
Merge branch 'branch-A-JavaDoc'

bbbbcccd committed:
Merge branch 'branch-A-CodingStandard'

bbbbcccd committed:
Fix coding standard

bbbbcccd committed:
Fix Level 9

NorbertLoh commented on src/main/java/command/ExitCommand.java:
```diff
@@ -0,0 +1,27 @@
+ */
+public class ExitCommand extends UserCommand {
+    @Override 
+    public boolean continueRunning() {
```
I like the use of verbs to represent method names.

NorbertLoh commented on src/main/java/task/Task.java:
```diff
@@ -0,0 +1,50 @@
+ */
+public abstract class Task {
+    private String description;
+    private boolean isDone;
```
I like how the name of booleans sounds like booleans.

NorbertLoh commented on src/main/java/task/TaskList.java:
```diff
@@ -0,0 +1,51 @@
+        this.taskList.add(t);
+    }
+
+    public int size() {
```
Perhaps this can sound more like a verb.
```suggestion
    public int getSize()
```

NorbertLoh reviewed (commented):
LGTM 👍. Just some nits to fix.

maximustan2 commented on src/main/java/command/UserCommand.java:
```diff
@@ -0,0 +1,57 @@
+     * Checks whether to continue running the program
+     * @return Boolean of whether the command is an exit command
+     */
+    public boolean continueRunning() {
```
It will be better if the method name can sound like a boolean, eg isContinueRunning

(ref: [see here](https://se-education.org/guides/conventions/java/index.html#:~:text=Boolean%20variables/methods%20should%20be%20named%20to%20sound%20like%20booleans))

maximustan2 commented on src/main/java/task/TaskList.java:
```diff
@@ -0,0 +1,51 @@
+        return this.taskList;
+    }
+
+    public Task get(int idx) throws IndexOutOfBoundsException {
```
Perhaps the method names can be made consistent, eg getTask as opposed to get

maximustan2 commented on src/test/java/command/UserCommandTest.java:
```diff
@@ -0,0 +1,45 @@
+        assertInstanceOf(ListCommand.class, UserCommand.toCommand("list"));
+    }
+
+    @Test
```
Perhaps the unit test can test if the resulting object possesses certain attributes rather than if its merely an instance of another

maximustan2 commented on src/main/java/task/Storage.java:
```diff
@@ -0,0 +1,88 @@
+     */
+    public ArrayList<Task> load() throws InvalidStorageFileException {
+        try {
+            ArrayList<Task> tasks = new ArrayList<>();
```
I like how the plural form is used as the variable name!

maximustan2 reviewed (commented):
LGTM! 👍 Just some nits

bbbbcccd committed:
Add Gui for chatbot

bbbbcccd committed:
Add chatbot response to the GUI

bbbbcccd committed:
Add .vscode folder into gitignore

bbbbcccd committed:
Update build.gradle to enableAssertions

bbbbcccd committed:
Add assertions to the execute method in commands

Assertions were added to check for any constraint
breaking errors that were not accounted for.

bbbbcccd committed:
Merge pull request #1 from bbbbcccd/branch-A-Assertions

Branch a assertions

bbbbcccd committed:
Extract createTask into createTodo, createDeadline, createEvent

bbbbcccd committed:
Align code quality to the expected standard

AddCommand.java: Extract functions to follow SLAP
Parser.java: Fix typo

bbbbcccd committed:
Merge pull request #2 from bbbbcccd/branch-A-CodeQuality

Branch a code quality

bbbbcccd committed:
Add GitHub actions workflow

bbbbcccd committed:
Add demo action

bbbbcccd committed:
Update compile path to java files

bbbbcccd committed:
Remove text-ui-test and add Junit tests

bbbbcccd committed:
Revert runtest.sh back and remove demo action

bbbbcccd committed:
Merge pull request #3 from bbbbcccd/branch-A-CI

Create CI using GitHub Actions

bbbbcccd committed:
Add method to parse input by tokens

bbbbcccd committed:
Add task type

bbbbcccd committed:
Use task type in storage line parser

bbbbcccd committed:
Provide generic methods to filter tasklist

bbbbcccd committed:
Provide options to find task by type

bbbbcccd committed:
Merge pull request #4 from bbbbcccd/branch-C-BetterSearch

Implement better search

bbbbcccd committed:
Extract storage file line parser

bbbbcccd committed:
Fix code quality

bbbbcccd committed:
Exit GUI window when user enters bye

bbbbcccd committed:
Merge branch 'master' of github.com:bbbbcccd/ip

bbbbcccd committed:
Update storage file path

bbbbcccd committed:
Add new storage file path into gitignore

bbbbcccd committed:
Add greeting message for GUI

bbbbcccd committed:
Remove CLI entry point

bbbbcccd committed:
Update images

bbbbcccd committed:
Add product screenshot

bbbbcccd committed:
Add user guide for LevelHundred

bbbbcccd committed:
Add CSS to GUI

bbbbcccd committed:
Update path to CSS file

bbbbcccd committed:
Update Gui screenshot

wes-w-z-h reviewed (commented):
LGTM! 👍

bbbbcccd committed:
Add comment

bbbbcccd committed:
Merge pull request #6 from bbbbcccd/branch-level-9

Add comment
